src/types/scriptProviderTypes.ts

```diff
@@ -43,7 +43,7 @@ export interface ScriptContextDerivedState
 }
 
 export interface ScriptProviderProps {
-    options: PayPalScriptOptions;
+    options?: PayPalScriptOptions;
```

Comment:
We need the `options` prop to be required since that's how `client-id` is passed through. The JS SDK will return a 400 status code if client-id is not provided as a query param.

```suggestion
    options: PayPalScriptOptions;
```

Reply:
as I recall this was needed to make one of the tests compile after converting it to typescript. I can look at that again; if the test that fails to compile is a demonstration of the requirement that you are referring to then I will look further into how to make that test valid without failing the type checking.